VulkanHppGenerator.hpp

```diff
@@ -212,11 +220,12 @@ class VulkanHppGenerator
 
   struct RequireData
   {
-    std::string              depends   = {};
-    std::vector<std::string> commands  = {};
-    std::vector<std::string> constants = {};
-    std::vector<std::string> types     = {};
-    int                      xmlLine   = {};
+    std::string                        depends       = {};
+    std::vector<std::string>           commands      = {};
+    std::map<std::string, std::string> enumConstants = {};
```

Comment:
Maybe change that to some `std::vector<NameValuePair>`, or so.

Reply:
I feel the `std::vector` semantics are a little weird here, and sort of breaking abstraction. Also, this means I have to define a `using NameValuePair = std::pair<...>` which also feels strange. 

If you insist, I can do it, but I'm personally not so sure.

Reply:
I tend to keep the order of elements as they are encountered in vk.xml. That's why there are just `std::vector`s here.
But as that is definitely not important for those values here, I'm fine with a `std::map` as well.